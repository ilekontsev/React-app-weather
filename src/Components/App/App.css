.root-wrap-app{
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    color: grey;
}