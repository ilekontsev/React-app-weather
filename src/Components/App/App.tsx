import React, { useEffect } from "react";
import { BrowserRouter, Route, Switch } from "react-router-dom";
import "./App.css";
import Header from "../Header/Header";
import BannerWeather from "../BannerWeather/BannerWeather";
import TodayTomorrow from "../TodayTomorrow/TodayTomorrow";
import SaveCity from "../SaveCity/SaveCity";
import WeekWeather from "../WeekWeather/WeekWeather";
import { useDispatch } from "react-redux";
import {
  actionSearchOfferVariant,
  actionSetPreservedCity,
} from "../../Store/Action";

const App = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    const cities = JSON.parse(localStorage.getItem("cities") || "[]");
    if (cities?.length !== 0) {
      dispatch(actionSetPreservedCity(cities));
    }
  }, []);

  const hiddenSearch = (e: React.MouseEvent) => {
    const target = e.target as Element;
    if (target.className !== "input-search-wrap") {
      dispatch(actionSearchOfferVariant([]));
    }
  };
  return (
    <BrowserRouter>
      <div
        data-testid={"block-tests-app"}
        className={"root-wrap-app"}
        onClick={hiddenSearch}
      >
        <Header />
        <BannerWeather />
        <Switch>
          <Route exact path="/" component={SaveCity} />
          <Route path="/today" component={TodayTomorrow} />
          <Route path="/tomorrow" component={TodayTomorrow} />
          <Route path="/week" component={WeekWeather} />
        </Switch>
      </div>
    </BrowserRouter>
  );
};

export default App;
