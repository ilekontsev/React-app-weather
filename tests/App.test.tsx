import React from "react";
import App from "../src/Components/App/App";

import { Provider } from "react-redux";
import store from "../src/Store/Store";
import ReactDOM from "react-dom";

describe(">>>APP", () => {
  let container: HTMLDivElement;
  // eslint-disable-next-line prefer-const
  container = document.createElement("div");

  afterEach(() => {
    document.body.removeChild(container);
    container.remove();
  });

  test("Check render class", () => {
    document.body.appendChild(container);
    ReactDOM.render(
      <Provider store={store}>
        <App />
      </Provider>,
      container
    );
    const findClass = document.querySelectorAll(".root-wrap-app");
    expect(findClass).toHaveLength(1);
  });
});
