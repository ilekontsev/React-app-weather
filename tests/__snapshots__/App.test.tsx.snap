// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>APP wda 1`] = `ShallowWrapper {}`;
